src/buffer.rs

```diff
@@ -243,6 +248,19 @@ impl Buffer {
         Ok(())
     }
 
+    /// Takes other buffer, measures its length and prints this buffer from the point where
+    /// the other stopped.
+    /// Used to implement autosuggestions.
+    pub fn print_rest<W>(&self, out: &mut W, other: &Buffer) -> io::Result<usize>
+        where W: Write
+    {
+        let matching_part = other.data.len();
+        let string: String = self.data.iter().skip(matching_part).cloned().collect();
+        out.write(string.as_bytes())?;
```

Comment:
@mmstick We cannot use method `as_bytes()` because we are dealing with `Vec<char>` and not `String`. I changed the signature to skip first x chars, but I haven't eliminated the heap allocation.